src/priority_fee.rs

```diff
@@ -96,7 +101,142 @@ pub struct PriorityFeeTracker {
+) -> Result<Vec<Pubkey>, TransactionValidationError> {
+    match transaction_meta {
+        None => Ok(Vec::with_capacity(0)),
+        Some(meta) => {
+            if meta.err.is_some() {
+                // statsd_count!("txn_failed", 1);
+                return Result::Err(TransactionValidationError::TransactionFailed);
+            }
+            let writable = meta
+                .loaded_writable_addresses
+                .into_iter()
+                .map(|v| Pubkey::try_from(v))
+                .filter(|p| p.is_ok())
+                .map(|p| p.unwrap())
+                .collect::<Vec<Pubkey>>();
+
+            Ok(writable)
+        }
+    }
+}
+fn extract_from_transaction(
+    mut transaction: SubscribeUpdateTransactionInfo,
+) -> Result<(Message, Vec<Pubkey>, bool), TransactionValidationError> {
+    let writable_accounts = extract_from_meta(transaction.meta.take())?;
+    let tran: &mut Transaction = transaction
+        .transaction
+        .as_mut()
+        .ok_or(TransactionValidationError::TransactionMissing)?;
+    let message: Message = tran
+        .message
+        .take()
+        .ok_or(TransactionValidationError::MessageMissing)?;
+    let is_vote = transaction.is_vote;
+
+    Result::Ok((message, writable_accounts, is_vote))
+}
+
+fn extract_from_message(
+    message: Message,
+) -> Result<
+    (Vec<Pubkey>, Vec<CompiledInstruction>, Option<MessageHeader>),
+    TransactionValidationError,
+> {
+    let account_keys = message.account_keys;
+    let compiled_instructions = message.instructions;
+
+    let accounts: Result<Vec<Pubkey>, _> = account_keys.into_iter().map(Pubkey::try_from).collect();
+    if let Err(_) = accounts {
+        return Err(TransactionValidationError::InvalidAccount);
+    }
+    let accounts = accounts.unwrap();
+
+    let compiled_instructions: Vec<CompiledInstruction> = compiled_instructions
+        .iter()
+        .map(|ix| {
+            CompiledInstruction::new_from_raw_parts(
+                ix.program_id_index as u8,
+                ix.data.clone(),
+                ix.accounts.clone(),
+            )
+        })
+        .collect();
+
+    Ok((accounts, compiled_instructions, message.header))
+}
+
+fn calculate_priority_fee_details(
+    accounts: &Vec<Pubkey>,
+    instructions: &Vec<CompiledInstruction>,
+    budget: &mut ComputeBudget,
+) -> Result<PrioritizationFeeDetails, TransactionError> {
+    let instructions_for_processing: HashMap<&Pubkey, &CompiledInstruction> = instructions
+        .iter()
+        .filter_map(|ix: &CompiledInstruction| {
+            let account = accounts.get(ix.program_id_index as usize);
+            if account.is_none() {
+                statsd_count!("program_id_index_not_found", 1);
+                return None;
+            }
+            Some((account.unwrap(), ix))
+        })
+        .collect();
+
+    budget.process_instructions(instructions_for_processing.into_iter(), true, true)
+}
+
+pub(crate) fn construct_writable_accounts<T>(
```

Comment:
Method that constructs the writable accounts. 
If header not there all accounts will be returned
Otherwise the logic will be applied